File: sum_a_list_but_ignore_any_duplicates.py
def sum_no_duplicates(l):
    dict = {}
    res=0
    for i in l:
        dict[i] = dict.get(i, 0) + 1
    for i,j in  dict.items():
        if j==1:
            res+=i
    return res

######################
def sum_no_duplicates(l):
    return sum(n for n in set(l) if l.count(n) == 1)

###########################3
def sum_no_duplicates(l):
    new = []
    for x in l:
        if l.count(x) == 1:
            new.append(x)
    return sum(new)
            

File: java/sum_a_list_but_ignore_any_duplicates.java
import java.util.*;

public class Solution{
    public static int sumNoDuplicates(int[] ar){
      //return Arrays.stream(ar).boxed().distinct().mapToInt(Integer::intValue).sum();
      TreeMap<Integer,Integer>mp=new TreeMap<>();
      for(Integer i:ar){
        mp.put(i,mp.containsKey(i)?mp.get(i)+1:1);
      }
      int sum=0;
      for(Map.Entry<Integer,Integer> e:mp.entrySet()){
        sum+=e.getValue().intValue()>1?0:e.getKey().intValue();
      }
      return sum;
    }
}

/////////////////
import java.util.Arrays;
import java.util.stream.Collectors;

public class Solution {
  public static int sumNoDuplicates(int[] arr) {
    return Arrays.stream(arr)
      .boxed()
      .collect(Collectors.groupingBy(x -> x, Collectors.counting()))
      .entrySet()
      .stream()
      .mapToInt(e -> e.getValue() == 1 ? e.getKey() : 0)
      .sum();
  }
}

///////////////
import static java.util.stream.Collectors.counting;
import static java.util.stream.Collectors.groupingBy;
import static java.util.stream.IntStream.of;

interface Solution {
  static int sumNoDuplicates(int[] arr) {
    return of(arr).boxed().collect(groupingBy(i -> i, counting())).entrySet()
            .stream().mapToInt(e -> e.getValue() > 1 ? 0 : e.getKey()).sum();
  }
}

///////////////
